node/client/src/lib.rs

```diff
@@ -567,3 +569,250 @@ impl sp_blockchain::HeaderBackend<Block> for Client {
 		}
 	}
 }
+
+/// Provides a `SignedPayload` for any runtime.
+///
+/// The first code block should set up all variables that are needed to create the
+/// `SignedPayload`. The second block can make use of the `SignedPayload`.
+///
+/// This is not done as a trait function since the return type depends on the runtime.
+/// This macro therefore uses the same approach as [`with_client!`].
+macro_rules! with_raw_payload {
```

Comment:
Why do we need a macro for this?

Reply:
The problem is that every runtime has their own `SignedExtra` and `SignedPayload`.  
These types do not share any trait, so I cannot use a trait instead.  
The `with_client` macro was done for a similar reason which did not work with a trait.  

I personally don't like it but did not see an easier solution.